### PR TITLE
feat: use interceptors to protect plugin's route

### PR DESCRIPTION
**Pull request by spacewander**
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible?

TODO: add doc

### CODE REVIEW

File: t/admin/plugin-metadata.t [hunk 1]
```diff
@@ -237,65 +237,217 @@ GET /t
...
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/plugin_metadata/prometheus',
+                ngx.HTTP_PUT,
+                [[{
+                    "interceptors": [
+                        {
+                        "name": "ip-restriction",
```
Comment (juzhiyuan) [inline t/admin/plugin-metadata.t:285]: would be better to add spaces here

Comment (spacewander) [reply t/admin/plugin-metadata.t:285]: @justinmclean 
Tweak

**Review by juzhiyuan — APPROVED** [review]
LGTM from codes style

**Review by membphis — APPROVED** [review]
Love this PR. Good design and implementation.

**Pull request merged**